### ISSUE TITLE
Repair Sim

### ISSUE THREAD

**Issue by zaxhutchinson** [zaxhutchinson/MAIA#16]
Opening this issue blind. I haven't tried to run a simulation after the ui_work merge but I'm 100% certain it is broken. I'll use this to document changes. Things I know need work are:

- Sim-to-loader interface: loader only stores dictionaries now, not game objects.
- Name changes: some of the names in the JSON have changed.
- Map drawing: I don't recall how the sim interfaces with the map canvas widget. I'm sure it needs work. There might be some synergy between the config map drawing.

The goal of the new branch related to this issue just to fix and bring sim and config in line. Once this is done. I'll start another branch to revamp the sim itself (which will probably trigger more ui changes).

**Comment by zaxhutchinson** [comment]
Work has been slow. In part this is because I've decided to completely overhaul how goals (gstates) work. This involves overhauling how winning and losing works.

I'm changing the entire thing to be points-based. Moving to a points system allows me to fix a bothersome issue in how goals are currently structured. Certain goals require the linking of a specific object used by a specific team in a specific map. This is a bit cumbersome to keep track of since it is something that has to be done out of the system.

- Maps will have a points-to-win value. Whichever team reaches or exceeds this value, wins.
- Points come from two sources: damaging other objects and triggers.
- Objects have two attributes that determine points awarded for damage: health and a damage-to-points scalar. Example: an object has a health of 100 and a scalar of 0.5. If another object causes 30 damage, they receive 15 points.
- Triggers belong to the map and allow for two meta conditions: 1. object at item and 2. item at item. This allows the system to limit triggers from being created if an item isn't present.

1. The "object" is agnostic. Any object. I'll leave for future work the ability to specify which object. I can think of scenarios where this might be desired, but I think they are second order (multi-agent team scenarios where only one object can "score").
1 and 2: Items can be restricted by the system to be only those items placed on the map. This means triggers can be created within the map admin screen. And I can do away with the Goal admin screen.

This also completely removes the ability to specify an xy location in a goal. Which is good. Because it will allow items to move and the trigger still work.

**Comment by zaxhutchinson** [comment]
To document some things:

- GStates have gone away and have been replaced by triggers. Triggers are stored in the maps.json file for each side. Triggers are side-specific. Triggers do not end the game like gstates did. They award points. Each side has a `points_to_win` value. If the team assigned to that side reaches this value it wins.
- I realized late that the mapconfig and the sim were drawing the map using different, opposing methods. These have been unified somewhat. They both use the same technique but (and this warrants more investigation) too much of it is local to each UI. At some point, we need to see if more functionality can be pushed back into ui_widgets.py.
- Ending the sim needs work.
